### PR TITLE
[csrng] Improve documentation

### PR DESCRIPTION
**Pull request by andreaskurth**
This PR improves the CSRNG documentation. It fixes typos, clarifies sentences, fixes references to technical names (such as modules and register fields), formats additional code terms with `<tt>`, and defines the *CC* acronym and adds a link to the relevant standard. The changes are described individually by each commit message.

### CODE REVIEW

File: hw/ip/csrng/doc/_index.md [hunk 1]
```diff
@@ -234,7 +235,7 @@ This section describes the application interface, which is required for performi
 Each CSRNG instance corresponds to a unique application interface port, which implements the application interface described here.
 Any hardware peripherals which require complete control of an instance may connect directly to a dedicated interface port.
 Meanwhile peripherals without any special requirements (i.e. personalization strings or non-FIPS-approved, fully-deterministic number sequences) may share access to an instance via the entropy distribution network (EDN) IP.
-The EDN's manage the instantiation and reseeding of CSRNG instances for general use-cases, providing either on-demand or timed-delivery entropy streams to hardware peripherals.
+The EDNs manage the instantiation and reseeding of CSRNG instances for general use-cases, providing either on-demand or timed-delivery entropy streams to hardware peripherals.
```
Comment (tjaychen) [inline hw/ip/csrng/doc/_index.md:238]: similar to `entropy_src` above, should we change this to `edns`

**Review by tjaychen — COMMENTED** [review]
generally looks good to me.
but there seems to be a mixed of use `module` and MODULE.  Should we pick one and stick with it?

File: hw/ip/csrng/doc/_index.md (hunk 1 above)
Comment (andreaskurth) [reply hw/ip/csrng/doc/_index.md:238]: EDN and CSRNG are abbreviations that have been introduced in this document, so I think it's meaningful to keep them. In contrast, "ENTROPY_SRC" is neither a module name nor an abbreviation nor the identifier used in the code, so I replaced it.

**Comment by andreaskurth** [comment]
> generally looks good to me. but there seems to be a mixed of use `module` and MODULE. Should we pick one and stick with it?

Thanks for your feedback, @tjaychen! As mentioned above, I suggest to refer to modules either with their full name (e.g., Cryptographically Secure Random Number Generator), the canonical abbreviation thereof (e.g., CSRNG), or the module designator used in the code (e.g., `csrng`). So I would replace "ENTROPY_SRC" with `entropy_src` but not CSRNG with `csrng`. Does that make sense?

File: hw/ip/csrng/doc/_index.md [hunk 2]
```diff
@@ -202,7 +203,7 @@ The state_db has the follow attributes shown in the following table:
   <tr>
     <td>415:160</td>
     <td>Key</td>
-    <td> Value required by NIST to be held in the state instance, and is of size <tt>SeedLen</tt>.
+    <td> Value required by NIST to be held in the state instance, and is of size <tt>KeyLen</tt>.
```
Comment (mwbranstad) [inline hw/ip/csrng/doc/_index.md:206]: Good catch.

File: hw/ip/csrng/doc/_index.md [hunk 3]
```diff
@@ -404,29 +405,29 @@ The actions performed by each command, as well as which flags are supported, are
...
 
 Once a command has been completed, successfully or unsuccessfully, the CSRNG responds with a single cycle pulse on the `csrng_rsp_ack` signal associated with the same application interface port.
-If the command is successful the `csrng_rsp_sts` signal will indicate the value 0 (`CSRNG_OK`) in the same cycle.
+If the command is successful, the `csrng_rsp_sts` signal will indicate the value 0 (`CSRNG_OK`) in the same cycle.
 Otherwise the application will receive the value 1 (`CSRNG_ERROR`) on the `csrng_rsp_sts` signal.
 A number of exception cases to be considered are enumerated in NIST SP 800-90A, and may include events such as:
 * Failure of the entropy source
 * Attempts to use an instance which has not been properly instantiated, or
-* Attempts to generate data when an instance has exceeded its maximum seed.life.
+* Attempts to generate data when an instance has exceeded its maximum seed life.
```
Comment (mwbranstad) [inline hw/ip/csrng/doc/_index.md:413]: Good catch.

File: hw/ip/csrng/doc/_index.md [hunk 4]
```diff
@@ -404,29 +405,29 @@ The actions performed by each command, as well as which flags are supported, are
...
 In such cases, a 32-bit exception message will be propagated to firmware via the `hw_exc_sts` register, and a `cs_hw_inst_exc` interrupt will be raised.
 
 #### Generated Bits (`genbits`) Interface
 
-In addition to the command response signals there is all the bus for returning the generated bits.
+In addition to the command response signals there is a bus for returning the generated bits.
```
Comment (mwbranstad) [inline hw/ip/csrng/doc/_index.md:418]: really odd typo, good cleanse.

File: hw/ip/csrng/doc/_index.md [hunk 5]
```diff
@@ -404,29 +405,29 @@ The actions performed by each command, as well as which flags are supported, are
...
 This 129-bit bus consists of 128-bits, `genbits_bus`, for the random bit sequence itself, along with a single bit flag, `genbits_fips`, indicating whether the bits were considered fully in accordance with FIPS standards.
 
 There are two cases when the sequence will not be FIPS compliant:
-- Early in the boot sequence, the `ENTROPY_SRC` generates a seed from the first 384 bits pulled from the noise source.
+- Early in the boot sequence, the `entropy_src` generates a seed from the first 384 bits pulled from the noise source.
```
Comment (mwbranstad) [inline hw/ip/csrng/doc/_index.md:422]: Same comment as before.

File: hw/ip/csrng/doc/_index.md [hunk 6]
```diff
@@ -404,29 +405,29 @@ The actions performed by each command, as well as which flags are supported, are
...
 This initial seed is tested to ensure some minimum quality for obfuscation use- cases, but this boot seed is not expected to be full-entropy nor do these health checks meet the 1024-bit requirement for start-up health checks required by NIST 800-90B.
 - If `flag0` is asserted during instantiation, the resulting DRBG instance will have a fully-deterministic seed, determined only by user input data.
 Such a seed will be created only using factory-entropy and will lack the physical-entropy required by NIST SP 800-90A, and thus this DRBG instance will not be FIPS compliant.
 
 #### Handshaking signals
 
 The application command signal `csrng_req_bus` is accompanied by a `csrng_valid_signal`, which is asserted by the requester when the command is valid.
-CSRNG may stall incoming commands by desserting the `csrng_req_ready` signal.
+CSRNG may stall incoming commands by deasserting the `csrng_req_ready` signal.
```
Comment (mwbranstad) [inline hw/ip/csrng/doc/_index.md:430]: Maybe should even be de-asserting.

File: hw/ip/csrng/doc/_index.md [hunk 7]
```diff
@@ -524,13 +525,13 @@ The following waveform shows an example of how the entropy source hardware inter
 
 ### Interrupts
 
-The `cs_cmd_req_done` interrupt will assert when a csrng command has been completed.
+The `cs_cmd_req_done` interrupt will assert when a CSRNG command has been completed.
 
-The `cs_entropy_req` interrupt will assert when csrng requests for entropy from ENTROPY_SRC.
+The `cs_entropy_req` interrupt will assert when CSRNG requests entropy from `entropy_src`.
```
Comment (mwbranstad) [inline hw/ip/csrng/doc/_index.md:530]: keep as caps.

File: hw/ip/csrng/doc/_index.md [hunk 8]
```diff
@@ -524,13 +525,13 @@ The following waveform shows an example of how the entropy source hardware inter
...
 
-The `cs_hw_inst_exc` interrupt will assert when any of the hardware-controlled CSRNG instances encounters an exception while executing a command, either due to errors on the command sequencing, or an exception within the `ENTROPY_SRC` IP.
+The `cs_hw_inst_exc` interrupt will assert when any of the hardware-controlled CSRNG instances encounters an exception while executing a command, either due to errors on the command sequencing, or an exception within the `entropy_src` IP.
```
Comment (mwbranstad) [inline hw/ip/csrng/doc/_index.md:532]: Same, keep as caps.

**Comment by andreaskurth** [comment]
Thanks for your reviews, @mwbranstad and @tjaychen.

As requested by @mwbranstad, I have dropped the commit that consistently spells the module name of the entropy source as `entropy_src`, so this PR no longer changes any of the names for the entropy source.

If I understood @tjaychen's comment correctly, it's probably worth homogenizing the use of acronyms (such as CSRNG) and module names as defined in the code (such as `csrng` or `entropy_src`) throughout the documentation at some point, but I think that is beyond the scope of this PR.

**Review by mwbranstad — APPROVED** [review]
LGTM

**Pull request merged**